### ISSUE TITLE
`AWS::EC2::PrefixList` not accepting valid IPv6 CIDR entries

### ISSUE THREAD

**Issue by jburnham** [aws-cloudformation/cfn-lint#4121]
### CloudFormation Lint Version

cfn-lint 1.34.1

### What operating system are you using?

Mac

### Describe the bug

The `Cidr` field for the `AWS::EC2::PrefixList` Entry entity seems to expect only[^1] `ipv4-network` however CloudFormation also accepts IPv6 CIDR entries when the PrefixList is using the IPv6 `AddressFamily`. I am getting the following lint failures with the example template.

```
E3002 '2400:cb00::/32' is not a 'ipv4-network'
template.yaml:22:11

E3002 '2606:4700::/32' is not a 'ipv4-network'
template.yaml:23:11
```

[^1]: https://github.com/aws-cloudformation/cfn-lint/blob/07057264e583f6ac9f2f83939a12415dea84518a/src/cfnlint/data/schemas/providers/us_east_1/aws-ec2-prefixlist.json#L8

### Expected behavior

No lint warnings/errors.

### Reproduction template

```yaml
---
AWSTemplateFormatVersion: '2010-09-09'

Resources:
  CloudflareIPv4PrefixList:
    Type: AWS::EC2::PrefixList
    Properties:
      PrefixListName: "com.example.global.ipv4.cloudflare"
      AddressFamily: "IPv4"
      MaxEntries: 30
      Entries:
        - Cidr: 173.245.48.0/20
        - Cidr: 103.21.244.0/22

  CloudflareIPv6PrefixList:
    Type: AWS::EC2::PrefixList
    Properties:
      PrefixListName: "com.example.global.ipv6.cloudflare"
      AddressFamily: "IPv6"
      MaxEntries: 15
      Entries:
        - Cidr: 2400:cb00::/32
        - Cidr: 2606:4700::/32
```
